cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedStreamSizeScheduleStore.java

```diff
@@ -249,11 +251,12 @@ public void apply() throws Exception {
 
             String rowKey = Bytes.toString(next.getRow());
             String[] splits = rowKey.split(":");
-            if (splits.length != 6) {
+            if (splits.length != 7) {
               continue;
             }
-            Id.Program program = Id.Program.from(splits[1], splits[2], ProgramType.valueOf(splits[3]), splits[4]);
-            SchedulableProgramType programType = SchedulableProgramType.valueOf(splits[3]);
+            ProgramId program = new ApplicationId(splits[1], splits[2],
```

Comment:
why do we need that? when one is constructing ApplicationId, he needs to be know about version. But when constructing ProgramId, we can simply take in ApplicationId. Otherwise all constructors need to have version string in the constructor. Not sure if that is good.